crates/worker/src/partition/mod.rs

```diff
@@ -178,11 +181,13 @@ where
             leadership_state,
             state_machine,
             max_command_batch_size,
-            partition_store: Some(partition_store),
+            partition_store,
```

Comment:
I now need to retain the reference in the `PartitionProcessor` struct so that I can pass clones to `SnapshotProducer`s when required.